src/clr-core/badge/badge.element.scss

```diff
@@ -12,59 +14,57 @@
   border-radius: $clr_baselineRem_0_416;
   font-size: $clr_baselineRem_0_416;
   padding: 0 $clr_baselineRem_4px;
-  margin-right: $clr_baselineRem_0_25;
   white-space: nowrap;
   text-align: center;
-
-  background-color: var(--clr-badge-background-color);
-  color: var(--clr-badge-color);
+  background: var(--background);
+  color: var(--color);
 }
 
 :host(:visited) {
-  --clr-badge-color: var(--clr-color-on-neutral-600);
+  --color: #{$clr-color-on-neutral-600};
```

Comment:
In depth description of global vs local vars is here https://github.com/vmware/clarity/issues/4061 but mainly to prevent components from relying on global state which can get really weird when apps start embedding other apps/components.

Reply:
My issue is reliance on the SASS variables. We should start considering these to be constants and consider how that impacts users that want to update the color palette instead of component variables at a micro level.

Reply:
Updated to use color vars with sass fallbacks.